test/one/profiler/test/TestProcess.java

```diff
@@ -0,0 +1,266 @@
+        ProcessBuilder pb = new ProcessBuilder(cmd).inheritIO();
+        if (test.output()) {
+            pb.redirectOutput(createTempFile("%out", null));
+        }
+        if (test.error()) {
+            pb.redirectError(createTempFile("%err", null));
+        }
+        if (!test.agentArgs().isEmpty() || test.jvmArgs().contains("-Djava.library.path=build/lib")) {
+            pb.redirectOutput(createTempFile("%pout", null));
+            pb.redirectError(createTempFile("%perr", null));
+        }
+
+        this.p = pb.start();
+        // Give the JVM some time to initialize
+        Thread.sleep(1500);
+    }
+
+    private void addArgs(List<String> cmd, String args) {
+        if (args != null && !args.isEmpty()) {
+            args = substituteFiles(args);
+            for (StringTokenizer st = new StringTokenizer(args, " "); st.hasMoreTokens(); ) {
+                cmd.add(st.nextToken());
+            }
+        }
+    }
+
+    private String substituteFiles(String s) {
+        Matcher m = filePattern.matcher(s);
+        if (!m.find()) {
+            return s;
+        }
+
+        StringBuffer sb = new StringBuffer();
+        do {
+            File f = createTempFile(m.group(1), m.group(2));
+            m.appendReplacement(sb, f.toString());
+        } while (m.find());
+
+        m.appendTail(sb);
+        return sb.toString();
+    }
+
+    private File createTempFile(String fileId, String suffix) {
+        try {
+            File f = File.createTempFile("ap-" + fileId.substring(1), suffix);
+            tmpFiles.put(fileId, f);
+            return f;
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void clearTempFiles() {
+        for (File file : tmpFiles.values()) {
+            file.delete();
+        }
+    }
+
+    @Override
+    public void close() {
+        p.destroy();
+        
+        try {
+            waitForExit(p, 20);
+        } catch (TimeoutException e) {
+            log.log(Level.WARNING, "Failed to terminate child process", e);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        } finally {
+            try {
+                if (retainLog == true) {
+                    String workingDirectory = System.getProperty("user.dir");
+                    File directory = new File(workingDirectory + "/build/test/logs/" + testName);
+                    directory.mkdirs();
+                    File sourceDirectory = (tmpFiles.get("%pout") != null) ? tmpFiles.get("%pout") : tmpFiles.get("%out");
+                    File destinationDirectory = new File(directory, "stdout");
+                    sourceDirectory.renameTo(destinationDirectory);   
+                    
+                    sourceDirectory = (tmpFiles.get("%perr") != null) ? tmpFiles.get("%perr") : tmpFiles.get("%err");
+                    destinationDirectory = new File(directory, "stderr");
```

Comment:
In general, it's discouraged to reuse variables for different things.
In this particular case, I wouldn't introduce intermediate variables at all.